compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -276,6 +278,28 @@ namespace std
                 getAllocator BT_MetaFullName =  [lt|.get_allocator()|]
                 getAllocator _ = mempty
 
+        -- copy constructor with allocator
+        allocatorCopyCtor alloc = [lt|
```

Comment:
If we combine this into one function called something like `allocatorCopyMoveCtors`, we can share a ton of the identical code and just have `copyFieldInit` and `moveFieldInit` to handle the differences between wrapping with `std::move` and not.

Perhaps something like this will help too:

```
withMove x = [lt|std::move(#{x})|] -- and for not moving, just use id
```

Reply:
Updated.

Reply:
The functions still look nearly identical to me. I was sort of expecting something more along the lines of this:

```
allocatorCtorBuilder argumentAdjuster = ...
allocatorCopyCtor = allocatorCtorBuilder id
allocatorMoveCtor = allocatorCtorBuilder withMove
    where withMove x = ...
```

Reply:
I've not sent the updated commit yet :)